scrapy/downloadermiddlewares/retry.py

```diff
@@ -48,7 +48,10 @@ def from_crawler(cls, crawler):
         return cls(crawler.settings)
 
     def process_response(self, request, response, spider):
-        if request.meta.get('dont_retry', False):
+        if (request.meta.get('dont_retry', False) or
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
```

Comment:
We could keep it back compatible by only taking `handle_httpstatus_all` into consider when there is a settings key, for instance "DONT_RETRY_ HANDLE_HTTPSTATUS_ALL = True"